projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -1095,22 +1095,10 @@ export class DatatableComponent<TRow extends Row = any>
       return;
     }
 
-    let idx: number;
-    const cols = this._internalColumns.map((c, i) => {
-      c = { ...c };
-
-      if (c.$$id === column.$$id) {
-        idx = i;
-        c.width = newValue;
-
-        // set this so we can force the column
-        // width distribution to be to this value
-        c.$$oldWidth = newValue;
-      }
-
-      return c;
-    });
-
+    const idx = this._internalColumns.indexOf(column);
+    const cols = this._internalColumns.map(col => ({ ...col }));
+    cols[idx].width = newValue;
+    cols[idx].$$oldWidth = newValue;
```

Comment:
I guess it would make sense to keep the comment, at least I wouldn't know in the aftermath why we set the current width twice:

```suggestion
    // set this so we can force the column
    // width distribution to be to this value
    cols[idx].$$oldWidth = newValue;
```

Reply:
comment restored